test/distributed/test_inductor_collectives.py

```diff
@@ -1340,13 +1341,11 @@ def func(inp, *, tag, ranks, group_size):
         assert counter.op_count == 3  # It generates 2 getattr to unpack the array
         assert same(out, correct)
 
+    # This doesn't work in all cases, and now we properly loudly error.
+    # See: https://github.com/pytorch/pytorch/issues/151240
+    # When differentiable funcols are implemented can revert.
+    @unittest.expectedFailure
```

Comment:
Can we reduce the test to only the part that fails? Or have an assertRaises there for the specific behavior we want to see fail.